### ISSUE TITLE
WebUI "No Package Published Yet"

### ISSUE THREAD

**Issue by psyhomb** [n4bb12/verdaccio-github-oauth-ui#6]
**Describe the bug**
After configuring plugin everything works as expected on the `npm` cli side, access and publish permissions per package and scope are enforced, but I don't see any published packages via WebUI, instead only `No Package Published Yet` message is shown.

This is not the case if I'm using `htpasswd` plugin, with this plugin everything works as expected on both sides.

**Expected behavior**
To see published packages via WebUI if proper access permissions are applied

CLI side
```
# npm whoami --registry $REGISTRY_URL
username1
#
```

Package access constraints (server configuration)
```
packages:
  'my-published-pkgs*':
    access: username1
    publish: username1
```

TL;DR
published packages can be installed via `npm` yet they are not visible via WebUI.

**Comment by n4bb12** [comment]
What version of verdaccio and the plugin are you using?

**Comment by psyhomb** [comment]
From what I can see the latest verdaccio version is [3.7.1](https://github.com/verdaccio/verdaccio/releases/latest), do you think upgrade will solve the issue?

```
$ verdaccio --version
3.6.0
$
```
```
$ npm list --depth=0 | grep verdaccio-github-oauth-ui
+-- verdaccio-github-oauth-ui@1.4.0
$
```

**Comment by n4bb12** [comment]
Yea, stuff changed in verdaccio 3.7.0 and the plugin version 1.4.0 accounts for those changes, see here https://github.com/n4bb12/verdaccio-github-oauth-ui/commit/7c741972ec4d06ffc1c001a4a0666c19f5c59b21.

There's a good chance it could fix your issue.

**Comment by n4bb12** [comment]
I'll update the peer dependencies.

**Comment by psyhomb** [comment]
I've upgraded verdaccio to the newest version and issue still persist.
```
$ verdaccio --version
3.7.1
$
```

**Comment by n4bb12** [comment]
From the root directory of the package you published, what is the output of this?
```
node -p 'require("./package.json").name'
```

**Comment by psyhomb** [comment]
I see, you are trying to check if package name that I use under packages section in verdaccio configuration file is the same as one I have published, and the answer is yes, like I already mentioned rules have been applied and everything is working from CLI when I'm using `npm` tool.

Also everything is working as expected on both sides (CLI and WebUI) if I use `htpasswd` plugin with same configuration options.

```
# head -3 package.json 
{
  "name": "my-published-pkgs",
  "version": "1.0.0",
...
#
```
```
# node -p 'require("./package.json").name'
my-published-pkgs
#
```

**Comment by psyhomb** [comment]
Package becomes visible only if I change permissions to:
```
'**':
    access: $authenticated
    publish: $authenticated
```

**Comment by n4bb12** [comment]
**TLDR**  
You are doing everything right.
This plugin doesn't support user filtering yet.

**It's optional**
As per the docs, group and user filters are not a standard feature but depend on the plugin:

> The security constraints remain on the shoulders of the plugin being used, by default verdaccio uses the htpasswd plugin. **If you use a different plugin the behaviour might be different.** https://verdaccio.org/docs/en/packages

See also the sample config:

> you can specify usernames/groupnames **(depending on your auth plugin)** and three keywords: "$all", "$anonymous", "$authenticated" https://github.com/verdaccio/verdaccio/blob/master/conf/full.yaml#L80

**What now?**
I'd like to support your use case, you'll just have to wait for it.
In the mean time you can fall back to using `"$all", "$anonymous", "$authenticated"`.

**Comment by psyhomb** [comment]
@n4bb12 thanks for detailed explanation, I'm looking forward to new feature but until then I'll probably fallback to using predifined groups or `htpasswd` plugin with custom users/groups.

**Comment by n4bb12** [comment]
The reason it works when installing, but not in the UI, is that  npm install passes an encrypted bearer token, whereas the UI uses a JWT.
Both requests enter Verdaccio through different endpoints and go through different middleware.

After some debugging, I found out that Verdaccio has a builtin username filter for package access. I don't even need to implement this. It just didn't work because the username was never set.

While the username is stored in localStorage, it is not actually used for authentication. Verdaccio expects the username to be inside the JWT.

Oddly, [the plugin already does this](https://github.com/n4bb12/verdaccio-github-oauth-ui/blob/master/src/server/plugin/CallbackMiddleware.ts#L43), following what the [flow types](https://github.com/verdaccio/verdaccio/blob/master/types/index.js#L62) expect to be passed in. This is just not what Verdaccio expects because it [actually takes an object](https://github.com/verdaccio/verdaccio/blob/533ea45b70020babbc53c88c01fb684035549cac/src/lib/auth.js#L287).

The whole thing ended up being much easier than I thought.

**Comment by psyhomb** [comment]
Just installed version [1.5.0](https://github.com/n4bb12/verdaccio-github-oauth-ui/releases/tag/v1.5.0) and now with changes you've made everything is working as expected!

Thank you once again!
